### PR TITLE
implements entry/redemption fees

### PR DESCRIPTION
**Pull request by vito-kovalione**

### CODE REVIEW

**Review by TonioMacaronio — COMMENTED** [review]
One question, should there be the events emitted when fees are changed?

File: programs/accountant/src/events.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+use anchor_lang::prelude::*;
+
+#[event]
+pub struct PerformanceFeeUpdatedEvent {
+    pub accountant_key: Pubkey, 
+    pub performance_fee: u64,
+}
```
[the remark this answers is not in the document]
Comment (vito-kovalione) [reply programs/accountant/src/events.rs:7]: don't we have a timestamp in the transaction data?

File: programs/accountant/src/state/generic_accountant.rs [hunk 2]
```diff
@@ -53,22 +64,31 @@ impl Accountant for GenericAccountant {
         )
     }
 
-    fn set_fee(&mut self, fee: u64) -> Result<()> {
+    fn set_performance_fee(&mut self, fee: u64) -> Result<()> {
```
[the remark this answers is not in the document]
Comment (vito-kovalione) [reply programs/accountant/src/state/generic_accountant.rs:67]: make sense, will add validation

File: programs/accountant/src/events.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (codinghistorian) [reply programs/accountant/src/events.rs:7]: true, No need to strictly use Clock sysvar timestamp.

**Review by codinghistorian — APPROVED** [review]

**Review by codinghistorian — APPROVED** [review]

**Pull request merged**